### PR TITLE
[MM-21749] [MM-22214] Add "Beta" and "Community" label

### PR DESCRIPTION
**Pull request by hanzei**
#### Summary
This PR allows adding a "Beta" and a "Community" label to plugin.

It currently looks like this: (JIRA is just an example)
Screenshot from 2020-02-01 09-29-10
Screenshot from 2020-02-01 09-29-14
Screenshot from 2020-02-01 09-29-18

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-21749
https://mattermost.atlassian.net/browse/MM-22214

### CODE REVIEW

**Comment by hanzei** [comment]
@asaadmahmood This is the label we talked about.

**Comment by asaadmahmood** [comment]
@hanzei I think we should just have a dimmer label, like the bot label for the beta tag. The blue labels are way too prominent.

**Comment by hanzei** [comment]
@asaadmahmood :+1: for a dimmer label. Do you have a suggestion?

**Comment by asaadmahmood** [comment]
We can just use what we have.
So just use the Badge component and if we can't use that, then just give it these clases:
Screenshot 2020-01-27 at 2 34 46 PM

However, since we need a fixed background and color, we can give it a background of:
```
background: rgba(61, 60, 64, 0.15)
color: #444;
```

Screenshot 2020-01-27 at 2 33 54 PM

**Comment by hanzei** [comment]
Screenshot from 2020-01-27 10-41-32
Screenshot from 2020-01-27 10-41-35

@asaadmahmood It looks very similar to the bot label with this setting. Would you be fine it I just re use the style of the bot label?

**Comment by asaadmahmood** [comment]
Yes, we should resuse the style of the bot label, but I gave the background and color because if we use the bot label directly without giving it an inline style, the theme may overwrite the background and the color, so try it with a theme and see if the bg and color are consistent.

Also, the local tag looks quite big compared to the bot label. So we can make them of similar size.

**Comment by hanzei** [comment]
I've applied the same background as the local label. This is how it looks like with a dark theme:
Screenshot from 2020-01-27 10-50-36
Screenshot from 2020-01-27 10-50-31

The background and color don't get overwritten by the theme.

**Comment by asaadmahmood** [comment]
@hanzei Can I have a full page screenshot?

**Comment by hanzei** [comment]
Screenshot from 2020-01-27 11-01-41

**Comment by asaadmahmood** [comment]
This looks weird.
I'll get a pull and look at it locally to fix the dark theme.

File: cmd/generator/add.go [hunk 1]
```diff
@@ -15,6 +15,10 @@ import (
 
 func init() {
 	generatorCmd.AddCommand(addCmd)
+
+	addCmd.Flags().Bool("beta", false, "Mark release as Beta")
+	addCmd.Flags().Bool("official", false, "Mark this plugin is maintanied by Mattermost")
```
Comment (lieut-data) [inline cmd/generator/add.go]:
```suggestion
	addCmd.Flags().Bool("official", false, "Mark this plugin as maintained by Mattermost")
```

File: cmd/generator/add.go [hunk 2]
```diff
@@ -15,6 +15,10 @@ import (
...
+	addCmd.Flags().Bool("community", false, "Mark this plugin is maintanied by the community")
```
Comment (lieut-data) [inline cmd/generator/add.go]:
```suggestion
	addCmd.Flags().Bool("community", false, "Mark this plugin as maintained by the community")
```

File: cmd/generator/add.go [hunk 3]
```diff
@@ -81,12 +103,21 @@ var addCmd = &cobra.Command{
 			return errors.Wrap(err, "failed to download plugin signature")
 		}
 
+		labels := []model.Label{}
```
Comment (lieut-data) [inline cmd/generator/add.go]: nit:
```suggestion
		var labels []model.Label
```

File: internal/model/label.go [hunk 4]
```diff
@@ -7,3 +7,20 @@ type Label struct {
...
 }
+
+var AllLabels = []Label{
+	CommunityLabel,
+	BetaLabel,
+}
+
+var CommunityLabel Label = Label{
+	Name:        "Community",
+	Description: "This plugin is maintained by the Open Source Community.",
```
Comment (hanzei) [inline internal/model/label.go]: @aaronrothschild Could you please confirm that this wording is fine

File: cmd/generator/add.go [hunk 5]
```diff
@@ -81,12 +103,21 @@ var addCmd = &cobra.Command{
...
+		if beta {
+			labels = append(labels, model.BetaLabel)
+		}
+
+		if official {
```
Comment (lieut-data) [inline cmd/generator/add.go]: Should this be `community`?

File: internal/model/label.go [hunk 6]
```diff
@@ -7,3 +7,20 @@ type Label struct {
...
+	URL:         "https://mattermost.com/pl/default-community-plugins",
+}
+
+var BetaLabel Label = Label{
+	Name:        "Beta",
+	Description: "This plugin is currently in Beta and it's not recommended in production use.",
```
Comment (hanzei) [inline internal/model/label.go]: @aaronrothschild Same here

File: internal/model/label.go [hunk 7]
```diff
@@ -7,3 +7,20 @@ type Label struct {
...
+
+var CommunityLabel Label = Label{
+	Name:        "Community",
+	Description: "This plugin is maintained by the Open Source Community",
+	URL:         "https://mattermost.com/pl/default-community-plugins",
+}
+
+var BetaLabel Label = Label{
+	Name:        "Beta",
+	Description: "This plugin is currently in Beta and it's not recommended in production use.",
```
Comment (lieut-data) [inline internal/model/label.go]:
```suggestion
	Description: "This plugin is currently in beta and is not recommended for use in production.",
```

File: cmd/generator/add.go (hunk 1 above)
Comment (hanzei) [reply cmd/generator/add.go]: I requested the review to soon :confused:

File: internal/model/label.go (hunk 7 above)
Comment (hanzei) [reply internal/model/label.go]: I've changed it to:
`This plugin is currently in Beta and is not recommended for use in production.`

`Beta` is written with a capital `B` in the documentation (e.g. in https://docs.mattermost.com/process/definitions.html) Hence, I would also use a capital `B` here.

File: cmd/generator/add.go (hunk 3 above)
Comment (hanzei) [reply cmd/generator/add.go]: I think there is a noticeable nuisance here: `var labels []model.Label` will render to `null` in json, while `labels := []model.Label{}` renders to `[]` in json. I'm slightly favouring the latter as `null` might be confusing to read. 1/5

Comment (lieut-data) [reply cmd/generator/add.go]: Ah, do we not `omitempty` for this case?

Comment (hanzei) [reply cmd/generator/add.go]: No, we don't (https://github.com/mattermost/mattermost-marketplace/blob/master/internal/model/plugin.go#L17). We also don't `omitempty` for any other field. Hence, I did not add it for consistency.

**Comment by hanzei** [comment]
@jfrerich Would you mind taking a look at this PR given your involvement into plugin reviews?

File: internal/model/label.go (hunk 4 above)
Comment (aaronrothschild) [reply internal/model/label.go]: Yes, this wording is good.

File: internal/model/label.go (hunk 6 above)
Comment (aaronrothschild) [reply internal/model/label.go]: @hanzei Both are good to go.

**Pull request merged**